### PR TITLE
[FLINK-6246] Fix generic type of OutputTag in operator Output

### PR DESCRIPTION
**Pull request by aljoscha**
R: @dawidwys

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java [hunk 1]
```diff
@@ -619,20 +619,20 @@ public void collect(StreamRecord<T> record) {
...
 
 		@Override
-		public <X> void collect(OutputTag<?> outputTag, StreamRecord<X> record) {
+		public <X> void collect(OutputTag<X> outputTag, StreamRecord<X> record) {
 			for (int i = 0; i < outputs.length - 1; i++) {
 				Output<StreamRecord<T>> output = outputs[i];
 
 				// due to side outputs, StreamRecords of varying types can pass through the broadcasting
 				// collector so we need to cast
 				@SuppressWarnings({"unchecked", "rawtypes"})
```
Comment (dawidwys) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java]: Suppression not needed anymore. Also the comment is not adequate, as we do not cast anything.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java [hunk 2]
```diff
@@ -619,20 +619,20 @@ public void collect(StreamRecord<T> record) {
...
-				StreamRecord<T> shallowCopy = (StreamRecord<T>) record.copy(record.getValue());
+				StreamRecord<X> shallowCopy = record.copy(record.getValue());
 				output.collect(outputTag, shallowCopy);
 			}
 
 			// don't copy for the last output
 			@SuppressWarnings({"unchecked", "rawtypes"})
```
Comment (dawidwys) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java]: Remove suppression.

File: flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/iteration/IterateExample.java [hunk 3]
```diff
@@ -107,7 +107,8 @@ public static void main(String[] args) throws Exception {
 		}
 
 		// execute the program
-		env.execute("Streaming Iteration Example");
+		System.out.println(env.getExecutionPlan());
```
Comment (dawidwys) [inline flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/iteration/IterateExample.java]: I think this change should be reverted.

Comment (aljoscha) [inline flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/iteration/IterateExample.java]: Oh no, I was playing around with something. 😱

**Comment by aljoscha** [comment]
@dawidwys thanks for reviewing so quickly. I pushed a commit to address your comments.

**Comment by dawidwys** [comment]
Now, I think it is good to merge ;)

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**